src/devtools-connector/devtools-arc-inspector.ts

```diff
@@ -93,6 +104,19 @@ class DevtoolsArcInspector implements ArcInspector {
       messageType: 'recipe-instantiated',
       messageBody: {slotConnections, activeRecipe}
     });
+
+    if (!this.arc.isSpeculative) this.updateParticleSet(particles);
+  }
+
+  private updateParticleSet(particles: Particle[]) {
+    const particleSources = [];
+    particles.forEach(particle => {
+      particleSources.push(particle.spec.implFile);
+    });
+    this.arcDevtoolsChannel.send({
+      messageType: 'register-particle-sources',
```

Comment:
maybe 'watch-particle-sources' instead? It seems to me to better describe what the message requests, what do you think?